src/test/java/com/teragrep/jai_02/tests/KeyStoreAccessImplTest.java

```diff
@@ -68,28 +71,84 @@ public static void prepare() {
         });
     }
 
-    public void save() {
+    @Test
+    public void saveAndVerifyTest() {
         Assertions.assertDoesNotThrow(() -> {
             ksa.saveKey(
                     userName,
                     userPassWord.toCharArray());
+
+            boolean authOk = ksa.verifyKey(
+                    userName,
+                    userPassWord.toCharArray());
+
+            Assertions.assertTrue(authOk);
         });
     }
 
-    public void verify() {
+    @Test
+    public void aliasAlreadyExistsTest() {
         Assertions.assertDoesNotThrow(() -> {
-            boolean authOk = ksa.verifyKey(
+            ksa.saveKey(
                     userName,
                     userPassWord.toCharArray());
+        });
 
-            Assertions.assertTrue(authOk);
+        IllegalArgumentException saveKeyThrownException = Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            ksa.saveKey(
+                    userName,
+                    userPassWord.toCharArray());
         });
+
+        Assertions.assertEquals("Alias for username <[" + userName + "]> already exists in KeyStore!",
+                saveKeyThrownException.getMessage());
     }
 
     @Test
-    public void saveAndVerifyTest() {
-        save();
-        verify();
+    public void loadNonExistingEntryTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            ksa.deleteKey(userName);
+        });
+
+        Assertions.assertThrows(InvalidKeyException.class, () -> {
+            ksa.loadKey(userName);
+        }, "Username <[" + userName + "]> was not found in the map!");
```

Comment:
The supplied message is printed when the test _fails_ so the logic of this message should be inverted, yes?

Reply:
fixed, added assertion for message as it was supposed to be in the first place.